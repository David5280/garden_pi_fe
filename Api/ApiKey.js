export const apiKey = 'c3b27e2f72d0496c26da9270019b8ea8'