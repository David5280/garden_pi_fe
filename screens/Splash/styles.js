import { StyleSheet, Dimensions } from 'react-native';


export default StyleSheet.create({
  container: {
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#73A686',
    
  },  
  textContainer: {
    borderWidth: 2,
    borderColor: '#A14550',
    backgroundColor: '#C6E6AB',
    width: Dimensions.get('window').width * 0.875,
    height: Dimensions.get('window').height * 0.50,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.8,
    alignItems: 'center',
    borderRadius: 10,
    justifyContent: 'space-between',
    padding: 20
  },
  splashHeader: {
    fontSize: 24,
    fontWeight: 'bold'
  },
  splashEnterBtn: {
    borderWidth: 2,
    borderColor: '#A14550',
    backgroundColor: '#FAF0E6',
    width: 200,
    height: 50,
    borderRadius: 10,
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center'
  },
  plantIcon: {
    width: 200,
    height: 200
  }
});

