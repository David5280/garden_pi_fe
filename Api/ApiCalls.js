import { apiKey } from './ApiKey'

export const fetchWeather = () => {
  return fetch(`https://api.darksky.net/forecast/${apiKey}/39.751048,-104.996659`)
  .then(res => res.json())
  .then(res => res)
  .catch(error => error.message)
}