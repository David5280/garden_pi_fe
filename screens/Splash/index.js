import React, { Component } from 'react';
import { 
  View, 
  TouchableOpacity, 
  Text, 
  ImageBackground, 
  Image,
  ActivityIndicator, 
  Dimensions } from 'react-native';
import { fetchWeather, fetchGarden, fetchGardenEnv } from '../../Api/ApiCalls';
import { Header } from '../../components';
import styles from './styles';

export default class Splash extends Component {
  state = {
    foreCast: null,
    garden: null,
    env: null,
    appLoaded: false
  };

  async componentDidMount() {
    await this.getWeather()
    await this.getGarden()
    await this.getEnv()
    this.setState({ appLoaded: true })
  };
  getWeather = async () => {
    await fetchWeather() 
    .then(weatherData => this.setState({foreCast: weatherData}))
  };
  getGarden = async () => {
    await fetchGarden()
    .then(gardenData => this.setState({garden: gardenData}))
  };

  getEnv = async () => {
    await fetchGardenEnv()
    .then(envData => this.setState({env: envData}))

  };

  onEnterPress = () => {
    this.props.navigation.navigate('Home', {
      foreCast: this.state.foreCast,
      env: this.state.env
    })
  };

  render() {
    return (
      <ImageBackground 
        source={require('../../assets/images/splash-background.jpg')}
        style={styles.container}
      >
        <View>
          <Header fontsize={55}/>
          <View style={styles.textContainer}>

            {!this.state.appLoaded &&
            <View style={{ alignItems: 'center', justifyContent: 'center', marginTop: 'auto', marginBottom: 'auto' }}>
              <Image  
                style={{ height: 150, width: 150 }}
                source={require('../../assets/images/beeBoi.gif')}
              />
            </View>
            }
            {
              this.state.appLoaded &&
              <TouchableOpacity 
              style={styles.splashEnterBtn}
              onPress={this.onEnterPress}
              >
              <Text>
                Enter your Garden...
              </Text>
            </TouchableOpacity>
            }

          </View>
        </View>
      </ImageBackground>
    )
  }
};