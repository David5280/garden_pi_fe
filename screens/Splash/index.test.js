import React from 'react';
import Splash from './index';
import 'react-native';
import renderer from 'react-test-renderer';
import * as constants from '../../Api/ApiCalls'

jest.mock("expo-camera", () => {
  return {}
});

test('renders correctly', () => {
  const tree = renderer.create(<Splash />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('getWeather should fetch current weather', async () => {
  constants.fetchWeather = jest.fn()
  constants.fetchWeather()
  expect(constants.fetchWeather).toHaveBeenCalled()
})

test('getGarden should fetch current garden', async () => {
  constants.fetchGarden = jest.fn()
  constants.fetchGarden()
  expect(constants.fetchGarden).toHaveBeenCalled()
})

test('getenv should fetch current garden environment', async () => {
  constants.fetchGardenEnv = jest.fn()
  constants.fetchGardenEnv()
  expect(constants.fetchGardenEnv).toHaveBeenCalled()
})

