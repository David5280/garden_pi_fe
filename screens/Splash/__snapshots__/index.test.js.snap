// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#73A686",
      "height": 1334,
      "justifyContent": "space-evenly",
      "width": 750,
    }
  }
>
  <Image
    source=""
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
        undefined,
      ]
    }
  />
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#E64C3C",
              "fontSize": 55,
            },
          ]
        }
      >
        G
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#FF9933",
              "fontSize": 55,
            },
          ]
        }
      >
        A
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#F1C431",
              "fontSize": 55,
            },
          ]
        }
      >
        R
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#187F03",
              "fontSize": 55,
            },
          ]
        }
      >
        D
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#3998DB",
              "fontSize": 55,
            },
          ]
        }
      >
        E
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#4b0082",
              "fontSize": 55,
            },
          ]
        }
      >
        N
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "lemondrop",
              "textShadowColor": "black",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            },
            Object {
              "color": "#7f00ff",
              "fontSize": 55,
            },
          ]
        }
      >
        Pi
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 667,
          "justifyContent": "space-between",
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "width": 656.25,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": "auto",
            "marginTop": "auto",
          }
        }
      >
        <Image
          source=""
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
